### PR TITLE
ci: add bitbot to manage issues and PRs

### PR DESCRIPTION
**Pull request by Xunzhuo**
Resolves: https://github.com/envoyproxy/gateway/issues/732

### CODE REVIEW

**Comment by zirain** [comment]
where does the bot host?

File: .github/workflows/kube-prow-bot.yaml
Comment (zirain) [inline .github/workflows/kube-prow-bot.yaml:63]: can we reuse OWNER file?

Comment (Xunzhuo) [reply .github/workflows/kube-prow-bot.yaml]: Yes. It sounds good and we can follow: https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md.  But I plan to work on next release.

**Comment by Xunzhuo** [comment]
https://github.com/Xunzhuo/kube-prow-bot/pull/5 

FYI.

File: .github/workflows/kube-prow-bot.yaml
Comment (Xunzhuo) [reply .github/workflows/kube-prow-bot.yaml]: This is already be done in prowox.

**Comment by Xunzhuo** [comment]
PLUGINS defined in env of workflows, it is the inclusion relationship between permissions.

File: OWNERS
Comment (Xunzhuo) [inline OWNERS:23]: We need to make sure reviewers list in the section.

**Comment by Xunzhuo** [comment]
As for https://envoyproxy.slack.com/archives/C03E6NHLESV/p1678388104708569, I think the /lgtm and /approve can solve it. We do not need to change the conditions to 2 reviews, we just need a /lgtm from reviewer and /approve from maintainers, and then we can merge the PR.

Take this as an example: https://github.com/mosn/layotto/pull/898. a /lgtm and an /approve and without /hold, can automatically merge a PR. 

@youngnick may be familiar with this, this kind of workflows are widely used in k8s community.

File: OWNERS
Comment (arkodg) [inline OWNERS:14]: this list needs to be updated

File: OWNERS
Comment (arkodg) [inline OWNERS:28]: please add `LanceEa` to this list and rm `sunjayBhatia` (because he has not confirmed yet)

**Review by arkodg — CHANGES_REQUESTED** [review]
the maintainers list needs to be updated

File: OWNERS
Comment (arkodg) [reply OWNERS:14]: we need to rm `alexgervais` and `danehans`

File: OWNERS
Comment (Xunzhuo) [reply OWNERS]: Done

File: OWNERS
Comment (Xunzhuo) [reply OWNERS:14]: Done

File: .github/workflows/bitbot.yaml
Comment (arkodg) [inline .github/workflows/bitbot.yaml:81]: can you elaborate on why this is needed ?
would prefer if we started off with minimal permissions

**Comment by arkodg** [comment]
cc @mattklein123 @phlax

File: .github/workflows/bitbot.yaml
Comment (Xunzhuo) [reply .github/workflows/bitbot.yaml:81]: Yes, I will try to minimize the permissions the action will be granted.

**Comment by Xunzhuo** [comment]
After: https://github.com/envoyproxy/gateway/pull/1156

**Comment by Xunzhuo** [comment]
/hold

I do not have strong opinions to push to use this bot written by me. Compared with prow itself, this bot can just do some simple issues/PRs managements. Prow can support more commands and do e2e tests, and it is widely used in k8s communities, which makes it more stable and powerful.

So I think before everything goes on, I think we should decide if we can just use prow directly. If so, we need a cluster to deploy prow and may need some helps from CNCF service desks.

**Comment by zirain** [comment]
IMHO, this is too complex to maintain, can we use prow bot from kubernetest/test-infra?

**Comment by Xunzhuo** [comment]
I will close this first, reopen it if we still need it.

**Pull request closed without merging**
